app/routes/agenda.js

```diff
@@ -27,10 +29,36 @@ export default class AgendaRoute extends Route {
       include: 'status',
     });
 
+    const agendaitems = await this.store.query('agendaitem', {
+      filter: {
+        agenda: {
+          id: agenda.id,
+        },
+      },
+      include: 'mandatees',
+    });
+    const notas = [];
+    const announcements = [];
+    for (const agendaitem of agendaitems.sortBy('number').toArray()) {
+      const type = await agendaitem.type;
+      if (type.uri === CONSTANTS.AGENDA_ITEM_TYPES.NOTA) {
+        notas.push(agendaitem);
+      } else {
+        announcements.push(agendaitem);
+      }
+    }
+
+    // Load agendaitem pieces
+    await Promise.all(
+      agendaitems.map(this.throttledLoadingService.loadPieces.perform)
```

Comment:
Any idea what the performance impact of loading this here will be for an agenda with 100+ agendaitems and even more pieces? 

I'm not sure, but I think there's a good reason we didn't load them in the model here at this stage. 

Furthermore, I'm not even sure this needs to be done at all. The story specified clearly that the `document-naming` service should load and provide all the information the `AgendaCheck` needs. Now, we're still loading it in the frontend, which is much slower.

Unless I'm misunderstanding this?